src/control/server/mgmt_pool.go

```diff
@@ -63,19 +67,31 @@ func (svc *mgmtSvc) PoolCreate(ctx context.Context, req *mgmtpb.PoolCreateReq) (
 		return nil, FaultPoolNvmeTooSmall(req.Nvmebytes, targetCount)
 	}
 
-	dresp, err := mi.CallDrpc(drpc.ModuleMgmt, drpc.MethodPoolCreate, req)
-	if err != nil {
-		return nil, err
-	}
+	try := 0
+	for {
+		dresp, err := mi.CallDrpc(drpc.ModuleMgmt, drpc.MethodPoolCreate, req)
+		if err != nil {
+			return nil, err
+		}
 
-	resp := &mgmtpb.PoolCreateResp{}
-	if err = proto.Unmarshal(dresp.Body, resp); err != nil {
-		return nil, errors.Wrap(err, "unmarshal PoolCreate response")
-	}
+		resp := &mgmtpb.PoolCreateResp{}
+		if err = proto.Unmarshal(dresp.Body, resp); err != nil {
+			return nil, errors.Wrap(err, "unmarshal PoolCreate response")
+		}
 
-	svc.log.Debugf("MgmtSvc.PoolCreate dispatch, resp:%+v\n", *resp)
+		svc.log.Debugf("MgmtSvc.PoolCreate dispatch, try %d, resp:%+v\n", try, *resp)
 
-	return resp, nil
+		switch drpc.DaosStatus(resp.GetStatus()) {
+		// retryable errors
+		case drpc.DaosGroupVersionMismatch:
+			try++
+			if try >= maxCreateRetries {
+				return resp, nil
+			}
+		default:
+			return resp, nil
+		}
+	}
```

Comment:
Might take advantage of ctx. At least, 3 tries without any backoff might still end up with -DER_GRPVER, because the system map distribution currently retries with a 3s backoff upon errors.

Reply:
Yeah, adding the context to our dRPC calls is on the list of updates I want to make, but I figured that this PR should be tightly-focused on handling the retry issue.

That having been said, I think it would be good to check `<-ctx.Done()` on each turn of the loop here.

What do you suggest as a value for `maxCreateRetries`?

Reply:
By taking advantage of ctx, I mean we are free to retry until ctx times out or cancels. I don't know a magic value for maxCreateRetries. :) But I wouldn't retry more than once every 3s.

Reply:
Hmm. Yes, this is probably a better pattern than setting an arbitrary number of retries. My concern was that I didn't want dmg to hang indefinitely waiting for a response in the event that there is some strange problem with the create. Maybe instead I should set a request timeout value.

Reply:
> Maybe instead I should set a request timeout value.

I knew! Implementing that timeout with ctx is probably the cleanest approach available at the moment.

Reply:
Yep. After thinking about this a bit more, I decided that the context cancellation criteria should be set by the RPC invoker (i.e. `control.CreatePool()`). We have a default timeout of 5 minutes in the control API, but each request type can customize the timeout as necessary. In discussions with @mchaarawi, it was discovered that we might want to set the pool create timeout dynamically based on the pool size, as the time to create seems to scale linearly with the SCM size with some recent patches.

I also set the retry backoff to 1.5s instead of 3, in order to improve responsiveness (imagine we get -DER_GRPVER just before the map distribution completes).